lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
Rails will use the language codes as you wrote them (`pt-BR` & `zh-CN` respectively). I went to find [a canonical example](https://github.com/svenfuchs/rails-i18n/blob/master/rails/locale/zh-CN.yml) just to be sure that our locale usage had not deviated from Rails standard and I had just forgotten.

I’ve written code to convert the language to a Rails locale (and I’ll push it momentarily), but now I’m wondering if just using the straight language could be what we want?

Thank you, I think I have a handle on those macros now, but at the same time, if I get it wrong, that may be why.